arclightning_backend/src/main.rs

```diff
@@ -0,0 +1,201 @@
+extern crate futures;
+extern crate hyper;
+#[macro_use]
+extern crate serde_derive;
+#[macro_use]
+extern crate serde_json;
+extern crate toml;
+
+use futures::future;
+
+use hyper::rt::{Future, Stream};
+use hyper::service::service_fn;
+use hyper::{Body, Method, Request, Response, Server, StatusCode};
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::PathBuf;
+use std::sync::{Arc, Mutex};
+
+// after this concept is further understood, will switch to 'Either'
+type ResponseFuture = Box<Future<Item = Response<Body>, Error = hyper::Error> + Send>;
+
+// using PartialEq for unit tests
+#[derive(Debug, Serialize, Deserialize, PartialEq, Clone)]
+// Using clone in a unit test atm.  Might not be necessary
+struct Game {
+    name: String,
+    description: String,
+    genres: Vec<String>,
+    thumbnail_path: PathBuf,
+    exe_path: PathBuf,
+}
+
+fn router(request: Request<Body>) -> ResponseFuture {
+    let mut response = Response::new(Body::empty());
+
+    match (request.method(), request.uri().path()) {
+        (&Method::GET, "/api/v1/list_games") => {
+            *response.body_mut() = Body::from("test");
+        }
+        _ => {
+            *response.status_mut() = StatusCode::NOT_FOUND;
+        }
+    }
+    Box::new(future::ok(response))
+}
+
+fn toml_to_hashmap(toml_filepath: PathBuf) -> HashMap<String, Game> {
+    let mut games_toml = String::new();
+    let mut file = File::open(&toml_filepath).expect("Could not find .toml file");
```

Comment:
If it's a test or main, it's fine to panic, but helper functions should not panic. Try changing the return type of this function to `Result<HashMap<String, Game>, io::Error>` and then using `?` instead of `unwrap` or `expect`.